### ISSUE TITLE
Delete the global RingBuffer database.

### ISSUE THREAD

**Issue by therealddx** [therealddx/midvec2-gui#5]
As part of implementing RAM transfer between nodes, the application instantiates a global RingBuffer database.

At application close, the application needs to free that memory.

Qt documentation suggests the following course of action:
_"We recommend that you connect clean-up code to the [aboutToQuit()](https://doc.qt.io/qt-5/qcoreapplication.html#aboutToQuit) signal, instead of putting it in your application's main() function. This is because, [entirely convincing technical reason]"_

In short:
- `connect` the `aboutToQuit` signal of the GUI's `QApplication`,
- to a callback function (likely, one in `main.cpp`) that `delete`s the RingBuffer database.

**Comment by therealddx** [comment]
Done; verified with a qDebug() trace. Worth noting is, this sets a precedent for the application-- 1) that library resources get destructed before GUI resources; and 2) that MainWindow::onApplicationClose is now "the" catch-all for destructing anything with application scope.
